internal/agent/device/spec/manager.go

```diff
@@ -260,26 +260,24 @@ func (s *manager) Rollback(ctx context.Context, opts ...RollbackOption) error {
 			return err
 		}
 		// set the desired spec as failed
-
 		s.queue.SetFailed(version)
 	}
 
-	// copy the current rendered spec to the desired rendered spec
-	// this will reconcile the device with the desired "rollback" state
+	// rollback on disk state current -> desired
 	if err := s.deviceReadWriter.CopyFile(s.currentPath, s.desiredPath); err != nil {
 		return fmt.Errorf("%w: copy %q to %q: %w", errors.ErrCopySpec, s.currentPath, s.desiredPath, err)
 	}
 
-	// update cache
-	desired, err := s.Read(Desired)
+	current, err := s.Read(Current)
```

Comment:
note: this isn’t technically a logical change, since after the rollback the desired and current states are the same but it could be confusing to read.